packages/realm-web/src/User.ts

```diff
@@ -96,6 +96,16 @@ export class User<
   /** @inheritdoc */
   public readonly apiKeys: ApiKeyAuth;
 
+  /**
+   * an extended property that may be set, removed or defined by others on an object
+   *  instance statically or as a dynamic getter function
+   * HTTP headers to be sent with mongoClient and functions requests
+   *
+   * @type {Headers}
+   * @memberof User
+   */
+  public headers?: Headers;
```

Comment:
This would be the first public and writable property in the API.
My initial though is that we would like this to be a write-only API, through a setter method (`setHeaders` or something ...) instead of a public writable property on `User`.